src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp

```diff
@@ -984,18 +984,58 @@ void prepare_primitive_fusing::fuse_simple_primitives(program &p) {
             size_t fused_idx = can_fuse_parents[0] ? 0 : 1;
             size_t peer_idx  = can_fuse_parents[0] ? 1 : 0;
 
-            int p1_pnum = p.get_processing_order().get_processing_number(parents[fused_idx].first);
-            int p2_pnum = p.get_processing_order().get_processing_number(parents[peer_idx].first);
-
-            auto p1_dt = parents[fused_idx].first->get_output_layout().data_type;
-            auto p2_dt = parents[peer_idx].first->get_output_layout().data_type;
-
-            if (can_fuse_parents[peer_idx] &&
-               ((p1_pnum < p2_pnum && p1_dt == p2_dt) || (data_type_traits::is_floating_point(p2_dt) && !data_type_traits::is_floating_point(p1_dt)))) {
-                // Swap in 2 cases:
-                // 1. Both branches have same data type. Select branch with lower processing number
-                // 2. Peer node has fp32 output type, but fused node - int8. In that case we have to fuse to the branch
-                // with fp32 out type to avoid fp32 blobs in the quantized graph.
+            auto can_swap_parents = [&]() -> bool {
+                // Swap in below two cases
+                //     1. Peer node has fp32 output type, but fused node - int8.
+                //         - In that case we have to fuse to the branch with fp32 out type to avoid fp32 blobs in the quantized graph.
+                //     2. Both branches have same data type. Select branch with lower processing number
+                if (can_fuse_parents[peer_idx]) {
+                    auto p1_pnum = p.get_processing_order().get_processing_number(parents[fused_idx].first);
+                    auto p2_pnum = p.get_processing_order().get_processing_number(parents[peer_idx].first);
+                    auto p1_dt = parents[fused_idx].first->get_output_layout().data_type;
+                    auto p2_dt = parents[peer_idx].first->get_output_layout().data_type;
+                    if (data_type_traits::is_floating_point(p2_dt) && !data_type_traits::is_floating_point(p1_dt)) {
+                        return true;
```

Comment:
shoudn't the residual fusion be considered first than this?

Reply:
In this code, the _first-if_ (comparing data type) has no method to return false, but the _second-if_ (comparing process order) has method to return false. what if the _second-if_ runs first and returns false, then it will not check the _first-if_ because it has already returned. However, originally if one of the two conditions is true, it becomes true. In other words, considering that the _first-if_ may return true even though the _second-if_ returns false, this execution order may be right.

But if it needs to reflect the priority among the two condition, I will try to use two boolean variables inside the lambda function like below.

`auto can_swap_parents = [&]() -> bool {`
`    bool cond1, cond2 = false;`
`    first-if {cond1 = true'}`
`    second-if {cond2 = true or false;}`
`    return cond1 | cond2;`
`};`

How do you think about this?

Reply:
Prioritized checking fusing history than data type and updated.